### ISSUE TITLE
Single-Elimination-bracket bye matches

### ISSUE THREAD

**Issue by Jell1231** [slashinfty/tournament-pairings#4]
When generating a single elimination bracket, the matches that have a bye are then inserted have no matches are generated.

So this is the gist of it. I am trying to visualize the bracket generated by the logic provided by this library.
I am experiencing an issue, unfortunately. The matches with a bye are generated in the next round, but this means that the new match is the new match 1. This is out of order with how the bracket would look like on paper. I understand that the path variable points to the new match. But in visualizing it would be way more helpful to create a match between the player that gets the bye, and an actual bye opponent. Then create the new matches accordingly.

Is this something that is possible or is this not in line with the vision you have for this library?
Hope to hear more about this!

**Comment by slashinfty** [comment]
(the issue was moved because this package is responsible for generating single elimination brackets (and all pairings))

This packet generates single elimination brackets that mimic [Challonge](https://challonge.com/tournament/bracket_generator) brackets. If you are attempting to visualize the brackets, it is an excellent tool.

For instance, a 10 player single elimination bracket is as follows:

```
[
  {
    round: 1,
    match: 1,
    player1: 8,
    player2: 9,
    win: { round: 2, match: 1 }
  },
  {
    round: 1,
    match: 2,
    player1: 7,
    player2: 10,
    win: { round: 2, match: 3 }
  },
  {
    round: 2,
    match: 1,
    player1: 1,
    player2: null,
    win: { round: 3, match: 1 }
  },
  {
    round: 2,
    match: 2,
    player1: 4,
    player2: 5,
    win: { round: 3, match: 1 }
  },
  {
    round: 2,
    match: 3,
    player1: 2,
    player2: null,
    win: { round: 3, match: 2 }
  },
  {
    round: 2,
    match: 4,
    player1: 3,
    player2: 6,
    win: { round: 3, match: 2 }
  },
  {
    round: 3,
    match: 1,
    player1: null,
    player2: null,
    win: { round: 4, match: 1 }
  },
  {
    round: 3,
    match: 2,
    player1: null,
    player2: null,
    win: { round: 4, match: 1 }
  },
  { round: 4, match: 1, player1: null, player2: null }
]
```

Visually:
![bracket](https://i.ibb.co/3FrsryY/Screenshot-20241206-080621.png)

Basically, I'm curious what it is you're looking for beyond what the package currently does. Do you want the match between 7 and 10 to be match 3 with empty matches for matches 2 and 4 in round 1? If you're trying to visualize the bracket or represent it graphically, it should be drawn based on paths, not match numbers.

**Comment by slashinfty** [comment]
Inactivity.
